src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -1026,68 +1026,88 @@ updates its progress:
 How to deal with asynchronous tasks in the front-end?
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-All executing and most recently finished asynchronous tasks are displayed on the
-"Backgroud-Tasks" menu.
-
-The front-end developer should provide a description, success message and error
-messages for each task on ``TaskManagerMessageService.messages``.
-This messages can make use of the task metadata to provide more personalized messages.
-
-When submitting an asynchronous task, the developer should provide a callback
-that will be automatically triggered after the execution of that task.
-This can be done by using the ``TaskManagerService.subscribe``.
-
-Most of the times, all we want to do after a task completes the execution, is
-displaying a notification message based on the execution result. The
-``NotificationService.notifyTask`` will use the messages from
-``TaskManagerMessageService`` to display a success / error message based on the
-execution result of a task.
+All executing and most recently finished asynchronous tasks are displayed on
+"Background-Tasks" and if finished on "Recent-Notifications" in the menu bar.
+For each task three prefixes, a suffix and error descriptions should be
+provided. This can be achieved by appending
+``TaskManagerMessageService.messages``. Each task has three different
+states - running, success and failure. To achieve consistency among all tasks
+and states each message consists of a prefix and a suffix.
+
+Message prefix
+  Ensures consistency among all tasks, it resembles what operation is
+  done. It consists of a list of three verbs for each different state f.e.
+  ``['create', 'Creating', 'Created']``.
+
+1. Failed messages always start with ``'Failed to '`` and should be continued
```

Comment:
when enumerating, these can be taken care of by the doc system, so using `#` would automatically be rendered as the proper number